packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -406,30 +406,43 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
 
         // check import.meta usage
         if (rawUrl === 'import.meta') {
-          const prop = source.slice(end, end + 4)
-          if (prop === '.hot') {
+          const hotSuffix = '.hot'
+          const envSuffix = '.env'
+          const acceptSuffix = '.accept'
+          const exportsSuffix = '.acceptExports'
+          const prop = source.slice(end, end + hotSuffix.length)
+          if (prop === hotSuffix) {
             hasHMR = true
-            const endHot = end + 4 + (source[end + 4] === '?' ? 1 : 0)
-            if (source.slice(endHot, endHot + 7) === '.accept') {
+            const endHot =
+              end +
+              hotSuffix.length +
+              (source[end + hotSuffix.length] === '?' ? 1 : 0)
+            if (
+              source.slice(endHot, endHot + acceptSuffix.length) ===
+              acceptSuffix
+            ) {
```

Comment:
I'm personally in favour of the code before. I can know that it's slicing to check `.accept` in one line, instead here I have to scroll/hover to look up what `acceptSuffix` stands for. It's also more words to read.

Usually I think there's benefit of crude implementations like this, so I'm voting to keep this as is for now 😬

Reply:
(Have similar thoughts of this for #12199 too)

Reply:
It's fine. Personally, I prefer the clean code which is more explicit.

Reply:
Sounds reasonable. Thanks for the explanation